main/zoe/api/zoe-contract-facet.md

```diff
@@ -4,23 +4,20 @@
 
 A Zoe Contract Facet is an API object for a running contract instance to access the Zoe state for that instance. A Zoe Contract Facet is accessed synchronously from within the contract, and usually is referred to in code as `zcf`. The contract instance is launched by `E(zoe).makeInstance`, and is given access to the `zcf` object during that launch. In the operation below, the `instanceHandle` is the handle for the running contract instance.
 
-## zcf.reallocate(offerHandles, newAmountKeywordRecords, sparseKeywords)
+## zcf.reallocate(offerHandles, newAmountKeywordRecords)
+
 - `offerHandles` <router-link to="/glossary/#handle">`{Array <Handle>}`</router-link>
 - `newAmountKeywordRecords` <router-link to="/zoe/api/records.html#amountkeywordrecord">`{Array <AmountKeywordRecord>}`</router-link>
-- `sparseKeywords` `{Array <String>}` sparseKeywords is an optional array of string keywords, which may be a subset of allKeywords.
 
 Instruct Zoe to try to reallocate payouts for the given `offerHandles`.  This will only succeed if the reallocation 1) conserves rights, and 2) is 'offer-safe' for all parties involved. This reallocation is partial, meaning that it applies only to
```

Comment:
I'd tighten this to:
"Instruct Zoe to try to reallocate payouts for the given `offerHandle`s. This succeeds only if the reallocation:
- Conserves rights.
-.Is 'offer-safe' for all involved parties.
This reallocation is partial, so it only applies to the amounts associated with passed in `offerHandle`s. We ensure that with each reallocation, rights are conserved and offer safety enforced for all offers. This is true even though it is a partial reallocation. Once these invariants are true, they stay true until changes happen."